### ISSUE TITLE
[icu] [python3] list of dependencies for Red Hat needs updating

### ISSUE THREAD

**Issue by sweemer** [microsoft/vcpkg#40670]
### Is your feature request related to a problem? Please describe.

The portfiles for both [icu](https://github.com/microsoft/vcpkg/blob/b6fcbeaa7f8f7729d625ba808f4bce7432b05072/ports/icu/portfile.cmake#L6-L7) and [python3](https://github.com/microsoft/vcpkg/blob/b6fcbeaa7f8f7729d625ba808f4bce7432b05072/ports/python3/portfile.cmake#L11-L12) contain the following instructions for installing dependencies:

```
On recent Red Hat and Fedora derivatives:
    sudo dnf install autoconf automake autoconf-archive
```

Out of these, I can only find `autoconf` and `automake` in the package manager for RHEL 8.6  - it seems that `autoconf-archive` is not available.

I eventually figured out how to get the `python3` port to build by downloading `autoconf-archive` from [here](https://gnu.mirror.constant.com/autoconf-archive/), building and installing manually, and then moving the installed files from `/usr/local/share/aclocal` to `/usr/share/aclocal` where vcpkg can find them.

I'm not sure whether this process is something that should be documented in the portfile, but I unfortunately am not aware of any better solution. In any case, the instructions should be updated to avoid giving the impression that `autoconf-archive` is available from the Red Hat package manager when it is not.

### Proposed solution

Perhaps the wording can be updated as follows:

```
On recent Red Hat and Fedora derivatives:
    sudo dnf install autoconf automake
    install autoconf-archive from https://gnu.mirror.constant.com/autoconf-archive/
```

**Comment by SchaichAlonso** [comment]
The ports shouldn't emit a notice. We invoke code that requires `autoconf-archive` to be installed in [vcpkg_configure_make](https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_configure_make), therefore, if we emit a diagnostic for RHEL, it should be `vcpkg_configure_make` that emits it, rather then the ports.

RHEL seems to actually be maintaining an `autoconf-archive` package in a [supplementory repository](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/8/html/package_manifest/codereadylinuxbuilder-repository#CodeReadyLinuxBuilder-repository) ... but [require users to pay them](https://www.redhat.com/en/store/linux-platforms) in order to get access to it 💯✍

Given the

    install autoconf-archive from https://gnu.mirror.constant.com/autoconf-archive/

instruction is quite non-trivial, I suggest adding a Section for RHEL to the [supported hosts document](https://learn.microsoft.com/en-us/vcpkg/concepts/supported-hosts) explaining the steps reqiuired to deploy a usable `autoconf-archive` onto RHEL systems, rathern then emiting a diagnostic.

**Comment by dg0yt** [comment]
> The ports shouldn't emit a notice. We invoke code that requires `autoconf-archive` to be installed in [vcpkg_configure_make](https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_configure_make),

This is wrong. vcpkg provides `autoconf-archive` only for Windows where it takes all the tools from MSYS2.
